### PR TITLE
Replace timestamp with block number

### LINKED ISSUE #857: Timestamps coming from js are too unreliable, switch to block number instead
Why is this issue relevant?
Timestamps are added to the signature requests from the sdk, but node timestamps are notoriously unreliable when converting from milliseconds -> nanoseconds, seconds causing Invalid Type errors through serde, as well as `Message too old` errors
What steps are required to resolve this?

Does this change the spec? HTTP, extrinsic, or storage? Is it breaking? Clearly describe the new interface.

### PR DESCRIPTION
**Pull request by JesseAbram**
Closes #857

### CODE REVIEW

**Comment by frankiebee** [comment]
💛

File: crates/client/src/client.rs [hunk 1]
```diff
@@ -435,13 +434,3 @@ async fn select_validator_from_subgroup(
     };
     Ok(address.clone())
 }
-
-#[cfg(feature = "full-client-wasm")]
-fn get_current_time() -> SystemTime {
-    use std::time::{Duration, UNIX_EPOCH};
-    UNIX_EPOCH + Duration::from_secs(js_sys::Date::now() as u64)
-}
-#[cfg(not(feature = "full-client-wasm"))]
```
Comment (ameba23) [inline crates/client/src/client.rs:444]: Ah good that we no longer need to worry about the system clock on different platforms here

File: crates/threshold-signature-server/src/validator/api.rs [hunk 2]
```diff
@@ -48,7 +48,7 @@ use crate::{
 #[derive(Debug, Deserialize, Serialize, Clone)]
 pub struct Keys {
     pub keys: Vec<String>,
-    pub timestamp: SystemTime,
+    pub block_number: BlockNumber,
```
Comment (ameba23) [inline crates/threshold-signature-server/src/validator/api.rs:51]: Ah so we also change this in the `sync_kvdb` payload

**Review by ameba23 — APPROVED** [review]
:1st_place_medal:

File: crates/threshold-signature-server/src/validator/tests.rs [hunk 3]
```diff
@@ -206,7 +208,7 @@ async fn test_sync_kvdb() {
     assert_eq!(result_4.status(), 500);
     assert_eq!(result_4.text().await.unwrap(), "Forbidden Key");
 
-    keys.timestamp = keys.timestamp.checked_sub(TIME_BUFFER).unwrap();
+    keys.block_number = keys.block_number + BLOCK_BUFFER + 10;
```
Comment (HCastano) [inline crates/threshold-signature-server/src/validator/tests.rs:211]: Why do we add an extra `10` here if we already have the `BLOCK_BUFFER`?

File: crates/threshold-signature-server/src/validator/api.rs [hunk 4]
```diff
@@ -235,6 +242,7 @@ pub async fn get_and_store_values(
     recip_server_info: ServerInfo<subxt::utils::AccountId32>,
     signer: &PairSigner<EntropyConfig, sr25519::Pair>,
     x25519_secret: &StaticSecret,
+    rpc: &LegacyRpcMethods<EntropyConfig>,
```
Comment (HCastano) [inline crates/threshold-signature-server/src/validator/api.rs:245]: Would be nice to not need to pass this in, but 🤷‍♂️

**Review by HCastano — APPROVED** [review]
👍

File: crates/threshold-signature-server/src/validator/api.rs (hunk 2 above)
Comment (JesseAbram) [reply crates/threshold-signature-server/src/validator/api.rs:51]: ya cuz it shares the check_stale function but I wouldn't worry to much, gonna need a whole re write when we go to threshold

File: crates/threshold-signature-server/src/validator/api.rs (hunk 4 above)
Comment (JesseAbram) [reply crates/threshold-signature-server/src/validator/api.rs:245]: ya but wouldn't worry about it, all this is gonna need a re write when we go to threshold

**Pull request merged**